server/src/main/java/io/crate/execution/ddl/tables/AlterTableClient.java

```diff
@@ -120,32 +139,45 @@ public AlterTableClient(ClusterService clusterService,
         this.sessions = sessions;
         this.indexScopedSettings = indexScopedSettings;
         this.logicalReplicationService = logicalReplicationService;
+        this.threadPool = threadPool;
     }
 
     public CompletableFuture<Long> addColumn(AddColumnRequest addColumnRequest) {
         var newReferences = addColumnRequest.references();
-        String subject = null;
+        String exceptionSubject = null;
+        String warning = null;
         if (addColumnRequest.pKeyIndices().isEmpty() == false) {
-            subject = "primary key";
-        } else if (addColumnRequest.references().stream().anyMatch(ref -> ref instanceof GeneratedReference)) {
-            subject = "generated";
+            exceptionSubject = "primary key";
+        } else if (newReferences.stream().anyMatch(ref -> ref instanceof GeneratedReference)) {
+            exceptionSubject = "generated";
         } else {
             for (Reference newRef : newReferences) {
                 if (newReferences.stream().anyMatch(r -> r.column().isChildOf(newRef.column()))
                     && newRef.columnPolicy().equals(ColumnPolicy.IGNORED)) {
-                    subject = "sub column to an OBJECT(IGNORED) parent";
+                    warning = "Adding a sub column to an OBJECT(IGNORED) parent may shade existing data of this column as the table isn't empty";
```

Comment:
```suggestion
                    warning = "Adding a sub column to an OBJECT(IGNORED) parent may shade existing data of this column as the table may not be empty";
```

We don't actually know if the table has any data or not in it, so the message might confuse people who create a table and then add the column before inserting any rows.

Reply:
> We don't actually know if the table has any data or not in it, so the message might confuse people who create a table and then add the column before inserting any rows.

We do now, we check the row count, see the `getRowCount` call. Thus the check and warning only happens when the table isn't empty.

Reply:
IIUC we don't do this check anymore?

Reply:
We still do https://github.com/crate/crate/pull/16930/files#diff-90c3e1d66eb24490b5eaded7ee3e9696038836c011142814148a0406a739a309R149.